src/traits.rs

```diff
@@ -84,6 +84,9 @@ pub unsafe trait Logger {
     /// Panics if already acquired in the current execution context. Otherwise it must never fail.
     fn acquire();
 
+    /// Block until host has read all pending data
+    fn flush();
```

Comment:
this should be `unsafe` and have the same safety requirement as `write`: MUST be called after `acquire` and before `release`